### DISCUSSION TITLE
Add camera capture input to image relation component

### DISCUSSION THREAD

**Discussion by gbolag** [directus/directus#21875] [Feature Requests]
### Summary

We now have to upload photos to the storage and pick a photo among many images. This feature will add a button for instant capture to the component interface. It will display a video stream from a mobile or computer camera when clicked. This can be captured when a button on the video is clicked, and the captured image is saved to the storage and linked to the database when the component save button is clicked.

### Basic Example

```vue
<template>
  <div>
    <button @click="handleCameraInput">Capture Image from Camera</button>
    <img v-if="image" :src="image" alt="Captured Image" />
  </div>
</template>

<script>
export default {
  data() {
    return {
      image: null
    };
  },
  methods: {
    handleCameraInput() {
      // Code to capture image from camera and set it as the selected image
      // This functionality would depend on the specific implementation and available libraries
      // For simplicity, let's assume setting a placeholder image
      this.image = 'camera_image_placeholder.jpg';
    }
  }
};
</script>
```

### Motivation

**Mobile Content Management**: With the ability to capture images directly from a mobile device's camera, users can quickly add new images to their Directus project while on the go. This is especially useful for field workers, photographers, or anyone who needs to upload images from their mobile devices.

**Real-time Image Capture:** Users can capture images in real-time using their device's camera, eliminating the need to upload images from a separate source. This can streamline workflows and make the content creation process more efficient.

**Visual Documentation:** In scenarios where visual documentation is required, such as capturing photos for inventory management, inspection reports, or field surveys, users can use the camera input to quickly capture and upload relevant images directly to the Directus project.

**User Engagement**: Adding a camera input can enhance user engagement by providing an intuitive and interactive way for users to contribute content to the Directus project. This can encourage more active participation and collaboration among project stakeholders.

**Integration with IoT Devices:** In applications involving Internet of Things (IoT) devices equipped with cameras, users can leverage the camera input feature to capture and upload images from these devices directly to the Directus project. This can be useful for monitoring and visualizing data collected by IoT devices.

## Feature Overview
The proposed feature aims to enhance the Directus platform by introducing a camera input option to the image relation component. This feature will allow users to capture images directly from their device's camera and associate them with specific records in the Directus project.

## Design Details
1. **User Interface Integration:**
   - The camera input option will be added to the image relation component within the Directus interface.
   - Users will see an additional button or icon labeled "Capture Image" next to the existing options for uploading images.
   - Clicking on the "Capture Image" button will open the device's camera interface within the Directus application.

2. **Camera Interface:**
   - The camera interface will provide users with options to capture images using their device's built-in camera.
   - Users will be able to preview the captured image before finalizing the upload.
   - Basic functionalities such as zoom, flash, and resolution settings may be included depending on device compatibility.

3. **Integration with Directus Image Relation:**
   - After capturing an image, users will have the option to associate it with specific records in the Directus project.
   - The image will be uploaded and stored within the Directus system, similar to images uploaded via traditional file upload methods.
   - Users can select one or multiple records to which the captured image should be linked.

4. **Metadata Handling:**
   - Metadata associated with the captured image, such as timestamp, location, and device information, may be automatically captured and stored along with the image.
   - Directus administrators can configure metadata handling options based on project requirements.

5. **Permissions and Security:**
   - Permissions for accessing the camera input feature will be managed through Directus role-based permissions.
   - Administrators can define which users or user roles have access to the camera input functionality.

## Usage Examples
1. **Field Operations:** A field technician using the Directus mobile app can quickly capture photos of equipment or assets during inspections and associate them with corresponding records in the Directus project.
2. **Content Creation:** A content creator managing a blog or website through Directus can use the camera input feature to capture and upload images directly from their device, streamlining the content creation process.
3. **Inventory Management:** A warehouse manager can use the camera input to capture images of new inventory items as they arrive, providing visual documentation that is automatically linked to inventory records in Directus.

## Corner Cases
- **Offline Usage:** The camera input feature should gracefully handle scenarios where users are offline, allowing them to capture images that are stored locally and uploaded once an internet connection is restored.
- **Image Quality:** Directus should provide options to adjust image quality settings to optimize file size and resolution based on project requirements and device capabilities.

## Terminology
- **Camera Input:** The feature allowing users to capture images directly from their device's camera within the Directus interface.
- **Image Relation Component:** The module within Directus used to associate images with specific records in the project.

### Must Have:
- Integration of camera input option within the Directus interface.
- Ability to capture images using the device's built-in camera.
- Option to preview captured images before finalizing the upload.
- Seamless association of captured images with specific records in the Directus project.

### Should Have:
- Basic functionalities such as zoom, flash, and resolution settings in the camera interface.
- Metadata handling to capture and store additional information (e.g., timestamp, location) along with the images.
- Role-based permissions management to control access to the camera input feature.

### Could Have:
- Offline functionality allowing users to capture images even when offline and upload them later when connectivity is restored.
- Customizable image quality settings to optimize file size and resolution.
- Compatibility with a wide range of devices and browsers to ensure a consistent user experience.

### Won't Have:
- Advanced image editing capabilities (e.g., cropping, filters) within the Directus interface.
- Integration with third-party image editing tools or services.
- Real-time image processing or analysis features (e.g., object recognition, facial recognition).

### Drawbacks
- **Implementation Cost:** Adding camera input functionality may increase the code size and complexity of the Directus interface, requiring significant development effort and maintenance.
- **User Space Implementation:** Integrating camera input at the user level might be challenging due to limitations in accessing device hardware and browser compatibility issues.
- **Impact on User Learning Curve:** Introducing a new feature like camera input may require additional training and documentation for Directus users to understand and utilize effectively.
- **Integration Challenges:** Integrating camera input with existing and planned features within Directus may pose compatibility issues or require significant adjustments to ensure seamless functionality.
- **Migration Challenges:** Implementing camera input as a breaking change could necessitate updates and modifications to existing Directus applications, potentially disrupting workflows and requiring additional resources for migration.

### Alternatives
- **File Upload Only:** One alternative is to stick with the existing file upload functionality in Directus without adding camera input. This would simplify the implementation and avoid potential complexities associated with integrating camera functionality.
- **Third-party Integrations:** Instead of building camera input functionality directly into Directus, users could leverage third-party integrations or plugins that offer similar capabilities. This approach would allow users to add camera input functionality as needed without directly modifying the core Directus codebase.
- **Manual Image Upload:** Users could manually capture images using their device's camera app and then upload them through the existing file upload interface in Directus. While this may require additional steps, it avoids the need for complex camera integration within Directus.
- **No Change:** Another alternative is to forego adding camera input altogether and focus on other features or improvements within Directus. This would eliminate the need for additional development effort and potential complications associated with integrating camera functionality.

### Adoption Strategy
- **Breaking Change Analysis:** Before implementing the proposal, it's essential to assess whether adding camera input functionality constitutes a breaking change for existing Directus developers. If significant modifications to existing codebases or workflows are required, developers may need to update their projects accordingly.
- **Migration Plan:** If the implementation of camera input is deemed to be a breaking change, a comprehensive migration plan should be developed to guide existing Directus developers through the transition. This plan could include documentation, tools, and resources to facilitate the migration process and minimize disruption.
- **Compatibility Considerations:** It's crucial to consider how the introduction of camera input functionality will affect other projects in the Directus ecosystem. Compatibility with existing plugins, extensions, and integrations should be evaluated to ensure seamless adoption without compromising the stability or functionality of these projects.
- **Communication and Support:** Effective communication and support channels should be established to assist Directus developers in understanding and adopting the new functionality. This may include documentation updates, community forums, and direct assistance from the Directus team to address any questions or concerns that arise during the adoption process.

### Unresolved Questions
- How will the camera input be integrated with the Directus image relation component?
- What specific libraries or methods will be used to capture images from the camera?
- Are there any limitations or compatibility issues with certain devices or browsers when using camera input?
- How will the captured images be stored or processed within the Directus system?

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
